### PR TITLE
(WIP) build and publish CLI to github package registry under a development channel 

### PR DESCRIPTION
**Pull request by acunniffe**
Goal: easier testing of PRs

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: "set up node.js"
+      uses: actions/setup-node@v1
+      with:
+        node-version: "12.x"
+        registry-url: https://npm.pkg.github.com/
+        scope: "@useoptic"
+    - name: "intial yarn install"
```
Comment (devdoshi) [inline .github/workflows/publish.yml:35]: Is there any reason it has to be done before bumping the versions?

File: package.json [hunk 2]
```diff
@@ -12,8 +12,10 @@
     "registry:start-background": "yarn run registry:start && wait-on http-get://localhost:4873",
     "registry:stop": "cd docker/private-npm-registry && docker-compose down",
     "publish-local": "OPTIC_PUBLISH_SCOPE=private node ./workspaces/scripts/publish.js",
+    "publish-github": "OPTIC_PUBLISH_SCOPE=github node ./workspaces/scripts/publish.js",
     "publish": "OPTIC_PUBLISH_SCOPE=public node ./workspaces/scripts/publish.js",
-    "bump": "node ./workspaces/scripts/bump.js"
+    "bump": "node ./workspaces/scripts/bump.js",
+    "alias-for-github": "node ./workspaces/scripts/alias-for-github.js"
```
Comment (devdoshi) [inline package.json:18]: I don't see this anywhere

File: workspaces/scripts/publish.js [hunk 3]
```diff
@@ -4,71 +4,84 @@ const path = require('path');
 
 const { workspaces } = packageJson;
 console.log({ workspaces });
-const isPrivatePublish = process.env.OPTIC_PUBLISH_SCOPE !== 'public'
+const isPrivatePublish = process.env.OPTIC_PUBLISH_SCOPE !== 'public';
+const registry =
+  process.env.OPTIC_PUBLISH_SCOPE === ('private' && 'http://localhost:4873') ||
```
Comment (devdoshi) [inline workspaces/scripts/publish.js:9]: I think the parenthesis are not in the right order here

File: .github/workflows/publish.yml (hunk 1 above)
Comment (acunniffe) [reply .github/workflows/publish.yml:35]: `fs-events` required for your script

**Pull request closed without merging**